frontend/packages/metal3-plugin/src/components/dashboard/inventory-card.tsx

```diff
@@ -0,0 +1,152 @@
+import { getNamespace } from '@console/shared';
+
+import { getHostStorage, getHostNICs, getHostCPU, getHostMachineName } from '../../selectors';
+import { getInventoryQueries, HostQuery, getHostQueryResultError } from './queries';
+
+const getResources = (namespace: string, machineName: string): FirehoseResource[] => [
+  {
+    isList: false,
+    namespace,
+    name: machineName,
+    kind: referenceForModel(MachineModel),
+    prop: 'machine',
+  },
+];
+
+export const InventoryCard: React.FC<InventoryCardProps> = ({
+  obj,
+  watchPrometheus,
+  stopWatchPrometheusQuery,
+  prometheusResults,
+  watchK8sResource,
+  stopWatchK8sResource,
+  resources,
+}) => {
+  const namespace = getNamespace(obj);
+  const machineName = getHostMachineName(obj);
+  const machineAddresses = _.get(resources.machine, 'data.status.addresses', []);
+  const machineIP = _.get(machineAddresses.find((addr) => addr.type === 'InternalIP'), 'address');
+
+  React.useEffect(() => {
+    const k8sResources = getResources(namespace, machineName);
+    k8sResources.forEach((r) => watchK8sResource(r));
+    return () => {
+      k8sResources.forEach((r) => stopWatchK8sResource(r));
+    };
+  }, [watchK8sResource, stopWatchK8sResource, namespace, machineName]);
+
+  React.useEffect(() => {
+    if (machineIP) {
+      const queries = getInventoryQueries(machineIP);
+      Object.keys(queries).forEach((key) => watchPrometheus(queries[key]));
+      return () => {
+        Object.keys(queries).forEach((key) => stopWatchPrometheusQuery(queries[key]));
+      };
+    }
+    return undefined;
+  }, [watchPrometheus, stopWatchPrometheusQuery, machineIP]);
+
+  const diskCount = getHostStorage(obj).length;
+  const nicCount = getHostNICs(obj).length;
+  const cpuCount = getHostCPU(obj).count || 0;
+
+  const queries = getInventoryQueries(machineIP);
+
+  const podResult = prometheusResults.getIn([queries[HostQuery.NUMBER_OF_PODS], 'result'], {});
+  const podError = getHostQueryResultError(podResult);
+  const podCount = _.get(podResult, 'data.result', []).length;
+
+  const fanResult = prometheusResults.getIn([queries[HostQuery.NUMBER_OF_FANS], 'result'], {});
+  const fanError = getHostQueryResultError(fanResult);
+  const fans = _.get(fanResult, 'data.result', []);
+  const fanCount = fans.length;
+  const fanNotOkCount = fans.filter((el) => el.metric.status !== 'ok').length;
+
+  const psuResult = prometheusResults.getIn([queries[HostQuery.NUMBER_OF_PSUS], 'result'], {});
+  const psuError = getHostQueryResultError(psuResult);
+  const psus = _.get(psuResult, 'data.result', []);
+  const psuCount = psus.length;
+  const psuNotOkCount = psus.filter((el) => el.metric.status !== 'ok').length;
+
+  return (
+    <DashboardCard>
+      <DashboardCardHeader>
+        <DashboardCardTitle>Inventory</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <InventoryItem
+          isLoading={!podResult}
```

Comment:
this will always be true, since you have `{}` as default value `const podResult = prometheusResults.getIn([queries[HostQuery.NUMBER_OF_PODS], 'result'], {});`

same for `fanResult` and `psuResult`

Reply:
done